addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java

```diff
@@ -116,12 +120,96 @@ public void init() {
         // All these first as they are used by other fields
         this.addTextField(
                 TAB_OPTIONS, FIELD_REPORT_NAME_PATTERN, reportParam.getReportNamePattern());
+
+        currentTemplateDir = reportParam.getTemplateDirectory();
         this.addFileSelectField(
                 TAB_OPTIONS,
                 FIELD_TEMPLATE_DIR,
-                new File(reportParam.getTemplateDirectory()),
+                new File(currentTemplateDir),
                 JFileChooser.DIRECTORIES_ONLY,
                 null);
+        ZapTextField templateDirField = (ZapTextField) this.getField(FIELD_TEMPLATE_DIR);
+        templateDirField
+                .getDocument()
+                .addDocumentListener(
+                        new DocumentListener() {
+
+                            boolean ignoreEvents = false;
+
+                            @Override
+                            public void insertUpdate(DocumentEvent e) {
+                                if (ignoreEvents) {
+                                    // Will reenter if we reset an invalid directory
+                                    return;
+                                }
+                                String selectedTemplateDir = templateDirField.getText();
+                                if (selectedTemplateDir.contentEquals(currentTemplateDir)) {
+                                    return;
+                                }
+
+                                File file = new File(selectedTemplateDir);
+
+                                EventQueue.invokeLater(
```

Comment:
This will still execute in the EDT, in which case it can be executed directly?

Reply:
Yeah, I get this stack trace when I try to do it directly:

java.lang.IllegalStateException: Attempt to mutate in notification
	at javax.swing.text.AbstractDocument.writeLock(AbstractDocument.java:1338) ~[?:1.8.0_252]
	at javax.swing.text.AbstractDocument.replace(AbstractDocument.java:658) ~[?:1.8.0_252]
	at javax.swing.text.JTextComponent.setText(JTextComponent.java:1669) ~[?:1.8.0_252]
	at org.zaproxy.addon.reports.ReportDialog$1.insertUpdate(ReportDialog.java:182) ~[?:?]
	at javax.swing.text.AbstractDocument.fireInsertUpdate(AbstractDocument.java:201) ~[?:1.8.0_252]
	at javax.swing.text.AbstractDocument.handleInsertString(AbstractDocument.java:748) ~[?:1.8.0_252]
	at javax.swing.text.AbstractDocument.insertString(AbstractDocument.java:707) ~[?:1.8.0_252]
	at javax.swing.text.PlainDocument.insertString(PlainDocument.java:130) ~[?:1.8.0_252]
	at javax.swing.text.AbstractDocument.replace(AbstractDocument.java:669) ~[?:1.8.0_252]
	at javax.swing.text.JTextComponent.setText(JTextComponent.java:1669) ~[?:1.8.0_252]
	at org.zaproxy.zap.view.StandardFieldsDialog$11.actionPerformed(StandardFieldsDialog.java:1649) ~[main/:?]
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022) ~[?:1.8.0_252]
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348) ~[?:1.8.0_252]
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402) ~[?:1.8.0_252]
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259) ~[?:1.8.0_252]
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252) ~[?:1.8.0_252]
	at java.awt.Component.processMouseEvent(Component.java:6539) ~[?:1.8.0_252]
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324) ~[?:1.8.0_252]
	at java.awt.Component.processEvent(Component.java:6304) ~[?:1.8.0_252]
	at java.awt.Container.processEvent(Container.java:2239) ~[?:1.8.0_252]
	at java.awt.Component.dispatchEventImpl(Component.java:4889) ~[?:1.8.0_252]
	at java.awt.Container.dispatchEventImpl(Container.java:2297) ~[?:1.8.0_252]
	at java.awt.Component.dispatchEvent(Component.java:4711) ~[?:1.8.0_252]
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904) ~[?:1.8.0_252]
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535) ~[?:1.8.0_252]
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476) ~[?:1.8.0_252]
	at java.awt.Container.dispatchEventImpl(Container.java:2283) ~[?:1.8.0_252]
	at java.awt.Window.dispatchEventImpl(Window.java:2746) ~[?:1.8.0_252]
	at java.awt.Component.dispatchEvent(Component.java:4711) ~[?:1.8.0_252]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760) ~[?:1.8.0_252]
	at java.awt.EventQueue.access$500(EventQueue.java:97) ~[?:1.8.0_252]
	at java.awt.EventQueue$3.run(EventQueue.java:709) ~[?:1.8.0_252]
	at java.awt.EventQueue$3.run(EventQueue.java:703) ~[?:1.8.0_252]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_252]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74) ~[?:1.8.0_252]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84) ~[?:1.8.0_252]
	at java.awt.EventQueue$4.run(EventQueue.java:733) ~[?:1.8.0_252]
	at java.awt.EventQueue$4.run(EventQueue.java:731) ~[?:1.8.0_252]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_252]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74) ~[?:1.8.0_252]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730) ~[?:1.8.0_252]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205) [?:1.8.0_252]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116) [?:1.8.0_252]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105) [?:1.8.0_252]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:1.8.0_252]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93) [?:1.8.0_252]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82) [?:1.8.0_252]